File: GoogleCodeJam/2021/ReverseSort.py
def min_index(l, start):
    mini = l[start]
    index = start
    i = start
    while i < len(l):
        if min(mini, l[i]) == l[i]:
            mini = l[i]
            index = i
        i += 1
    return index


def reverse(l, start, end):
    r = []
    i = end
    while i > start - 1:
        r.append(l[i])
        i -= 1
    return r


tests = int(input())
for casenum in range(tests):
    n = int(input())
    l = list(map((lambda x: int(x)), str(input()).split(" ")))
    cost = 0
    for i in range(n):
        j = min_index(l, i)
        cost += (j - i + 1)
        r = reverse(l, i, j)
        l = l[:i] + r + l[(j + 1):]
    print("Case #" + str(casenum + 1) + ": " + str(cost))

File: GoogleCodeJam/2021/ReverseSort.java
import java.util.Scanner;

class ReverseSort {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int tests = sc.nextInt();
        for(int casenum = 1; casenum < tests + 1; casenum++){
            int n = sc.nextInt();
            int[] l = new int[n];
            for(int i = 0; i < n; i++){
                l[i] = sc.nextInt();
            }
            int cost = 0;
            for(int i = 0; i < n - 1; i++){
                int j = minIndex(l, i);
                cost += (j - i + 1);
                int[] r = reverse(l, i, j);
                int index = 0;
                for(int s = i; s < j + 1; s++){
                    l[s] = r[index];
                    index++;
                }
            }
            System.out.println("Case #" + casenum + ": " + cost);
        }
    }
    public static int minIndex(int[] l, int start){
        int min = l[start];
        int index = start;
        for(int i = start; i < l.length; i++){
            if(Math.min(min, l[i]) == l[i]){
                min = l[i];
                index = i;
            }
        }
        return index;
    }
    public static int[] reverse(int[] l, int start, int end){
        int[] r = new int[end - start + 1];
        int index = 0;
        for(int i = end; i > start - 1; i--){
            r[index] = l[i];
            index++;
        }
        return r;
    }
}
